src/libraries/System.Net.Requests/src/System/Net/HttpWebRequest.cs

```diff
@@ -1066,7 +1070,12 @@ private Task<Stream> InternalGetRequestStream()
                 throw new InvalidOperationException(SR.net_reqsubmitted);
             }
 
-            _requestStream = new RequestStream();
+            // Create stream buffer for transferring data from RequestStream to the StreamContent.
+            StreamBuffer streamBuffer = new StreamBuffer();
+            _sendRequestTask = SendRequest(async, new StreamContent(new HttpClientContentStream(streamBuffer)));
```

Comment:
A subtle change here is that we're no longer sending the `Content-Length` header when buffering if we go through the `StreamContent`.
If we change this to not open the connection until GetResponse is called, we should set it manually on the content.